node/src/Pos/Block/Network/Logic.hs

```diff
@@ -348,6 +342,28 @@ handleRequestedHeaders cont inRecovery headers = do
             logDebug msg
             throwM $ DialogUnexpected msg
   where
+    newestHeader = headers ^. _NewestFirst . _neHead
+    oldestHeader = headers ^. _NewestFirst . _neLast
+    newestHash = headerHash newestHeader
+    oldestHash = headerHash oldestHeader
+    oldestEpoch = oldestHeader ^. epochIndexL
+
+    tryCalculateLrc = do
+        tipEpoch <- view epochIndexL <$> DB.getTipHeader @ssc
+        if oldestEpoch == tipEpoch + 1 then do
+            logDebug "handleRequesteHeaders: LRC started"
+            let handler UnknownBlocksForLrc =
+                    logWarning $
+                    "handleRequestedHeaders: tried lrcSingleShot, " <>
+                    "got UnknownBlocksForLrc"
+                handler e                   = throwM e
+            withStateLockNoMetrics LowPriority $ const $
```

Comment:
IMO, it should be `HighPriority`, because it's block processing.

Reply:
Fair point.